### PR TITLE
[exporter/awsprometheusremotewriteexporter] enable gocritic

### PR DESCRIPTION
**Pull request by my-git9**
**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10466

Enable gocritic in exporter/awsprometheusremotewriteexporter

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

File: exporter/awsprometheusremotewriteexporter/factory_test.go [hunk 1]
```diff
@@ -33,18 +32,18 @@ func TestType(t *testing.T) {
...
 	assert.NotNil(t, cfg, "failed to create default config")
 	assert.NoError(t, configtest.CheckConfigStruct(cfg))
 }
 
-//Tests whether or not a correct Metrics Exporter from the default Config parameters
+// Tests whether or not a correct Metrics Exporter from the default Config parameters
 func TestCreateMetricsExporter(t *testing.T) {
 	af := NewFactory()
-	validConfigWithAuth := af.CreateDefaultConfig().(*Config)
+	validConfigWithAuth := af.CreateDefaultConfig()
```
Comment (djaglowski) [inline exporter/awsprometheusremotewriteexporter/factory_test.go:46]: It appears these assertions are necessary:
```
Error: .\factory_test.go:47:21: validConfigWithAuth.AuthConfig undefined (type config.Exporter has no field or method AuthConfig)
Error: .\factory_test.go:56:23: invalidConfigWithAuth.AuthConfig undefined (type config.Exporter has no field or method AuthConfig)
Error: .\factory_test.go:59:15: invalidConfig.HTTPClientSettings undefined (type config.Exporter has no field or method HTTPClientSettings)
Error: .\factory_test.go:62:18: invalidTLSConfig.HTTPClientSettings undefined (type config.Exporter has no field or method HTTPClientSettings)
```

**Review by gouthamve — APPROVED** [review]
LGTM!

**Review by djaglowski — APPROVED** [review]

**Comment by my-git9** [comment]
@djaglowski Hi, I want to apply become a member of opentelemetry community,can you sponsored me?

**Pull request merged**